src/main/default/lwc/descriptionStrengthIndicator/descriptionStrengthIndicator.js

```diff
@@ -0,0 +1,115 @@
+import { LightningElement, api } from 'lwc';
+import thankYou from '@salesforce/label/c.cookbook_ThankYou';
+import provideDetails from '@salesforce/label/c.cookbook_ProvideDetails';
+import provideMoreDetails from '@salesforce/label/c.cookbook_ProvideMoreDetails';
+import dontKnowWhatToWrite from '@salesforce/label/c.cookbook_DontKnowWhatToWrite';
+import descriptionHintOne from '@salesforce/label/c.cookbook_DescriptionHintOne';
+import descriptionHintTwo from '@salesforce/label/c.cookbook_DescriptionHintTwo';
+import descriptionHintThree from '@salesforce/label/c.cookbook_DescriptionHintThree';
+
+/**
+ * The `descriptionStrengthIndicator` component is a dynamic indicator that shows the user if his case description has enough details so far. It also give hints to the user as to what information they show include in the description.
+ * @example
+ * <c-description-strength-indicator initial-message="Provide details" help-label="Don't know what to write?" keep-going-message="Provide more details" keep-going-threshold="75" final-message="Thank you!"></c-description-strength-indicator>
+ */
+export default class DescriptionStrengthIndicator extends LightningElement {
+  labels = {
+    thankYou,
+    provideDetails,
+    provideMoreDetails,
+    dontKnowWhatToWrite,
+    descriptionHintOne,
+    descriptionHintTwo,
+    descriptionHintThree
+  };
+
+  /**
+   * The label to be shown to the user.
+   * @type {string}
+   * @defaultValue `'Don't know what to write?'`
+   */
+  @api helpLabel = this.labels.dontKnowWhatToWrite;
+
+  /**
+   * The initial message to be shown to the user.
+   * @type {string}
+   * @defaultValue `'Provide details'`
+   */
+  @api initialMessage = this.labels.provideDetails;
+
+  /**
+   * The message to be shown to encourage the user to keep going.
+   * @type {string}
+   * @defaultValue `'Provide more details'`
+   */
+  @api keepGoingMessage = this.labels.provideMoreDetails;
+
+  /**
+   * The progress value from where the keepGoingMessage will be shown.
+   * @type {number}
+   * @defaultValue `75`
+   */
+  @api keepGoingThreshold = 75;
+
+  /**
+   * The final message to be shown to the user when the progress indicator is full.
+   * @type {string}
+   * @defaultValue `'Thank you!'`
+   */
+  @api finalMessage = this.labels.thankYou;
+
+  /** @type {number} */
+  _progress = 0;
+
+  /**
+   * Returns the message to be shown to the user.
+   * @returns {string}
+   */
+  get message() {
+    if (this._progress === 100) {
```

Comment:
I would perhaps extract these into other functions?
`get isCompleted() { return this._progress == 100 }`

Reply:
I have already implemented a `get isFull()` so yeah I can reuse it here with other functions for the keepGoing state and use it here to make the code more readable.👌